src/components/ProductList/ItemsList/ItemsList.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import ProductItem from '../../MostPopular/ProductItem';
+import HttpService from '../../../service/HttpService/httpService';
+import mockedData from '../../../mockedDataForTests';
+
+import './ItemsList.scss';
+
+const CN = 'items-container';
+const userAPI = new HttpService();
+const { mockedProductList } = mockedData;
+
+class ItemsList extends React.Component {
+  componentDidMount() {
+    this.getListItems();
+  }
+
+  async getListItems() {
+    const { onGetProductsSuccess, onGetProductsError, categoryName } = this.props;
+
+    try {
+      const response = await userAPI.get(`${process.env.BASE_URL}/${categoryName}`);
+      if (response.status === 404) {
+        throw Error(response.statusText);
+      }
+      onGetProductsSuccess(response.data);
+    } catch (error) {
+      onGetProductsError(error);
+    }
+  }
+
+  getItems() {
+    const {
+      itemsOnPage, ascendingOrder, orderType, itemList, error
+    } = this.props;
+
+    if (error) {
+      return (
+        <div className={`${CN}__error-container`}>
+          Please run npm run mock:api to start the server and reload the page, or check the URL of your request.
+        </div>
+      );
+    }
+
+    const itemsToRender = [...itemList];
+
+    const sortByItemName = (prevEl, nextEl) => {
+      const prevName = prevEl.title.toUpperCase();
+      const nextName = nextEl.title.toUpperCase();
+      const isPrevLetterBigger = (prevName > nextName) ? 1 : 0;
+
+      if (ascendingOrder) {
+        return (prevName < nextName) ? -1 : isPrevLetterBigger;
+      }
+      return (prevName < nextName) ? isPrevLetterBigger : -1;
+    };
+
+    switch (orderType) {
+      case 'Price':
+        itemsToRender.sort((prevEl, nextEl) => (ascendingOrder ? prevEl.price - nextEl.price : nextEl.price - prevEl.price));
+        break;
+      case 'Name':
+        itemsToRender.sort((prevEl, nextEl) => sortByItemName(prevEl, nextEl));
+        break;
+      case 'Position':
+        if (!ascendingOrder) {
+          itemsToRender.reverse();
+        }
+        break;
+      default:
+        break;
+    }
+
+    return itemsToRender.slice(0, itemsOnPage).map((el) => (
+      <ProductItem product={el} key={el.title} />
+    ));
+  }
+
+  render() {
+    return (
+      <div className={CN}>{this.getItems()}</div>
+    );
+  }
+}
+
+ItemsList.propTypes = {
+  ascendingOrder: PropTypes.bool,
+  categoryName: PropTypes.string,
+  itemList: PropTypes.array,
+  itemsOnPage: PropTypes.number,
+  onGetProductsError: PropTypes.func,
+  onGetProductsSuccess: PropTypes.func
+};
+
+ItemsList.defaultProps = {
+  ascendingOrder: true,
+  categoryName: 'product-list',
+  itemList: mockedProductList,
+  itemsOnPage: 6,
+  onGetProductsError: null,
+  onGetProductsSuccess: null
+};
+
+export default ItemsList;
```

Comment:
Also, I don't think that you need class component right there

Reply:
I am using ComponentDidMount method, that's why it is a smart component. Should I rewrite this to functional component?

Reply:
Well, yes. Try to use class component whenever you need to use state.
But ok, you can leave this one as is.
Just take into account for the future